### PR TITLE
feat: Implement process member bundle request queue

### PR DESCRIPTION
**Pull request by levansuper**
closes: #107

### CODE REVIEW

File: src/backgroundQueue/sinks/example.sink.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 // eslint-disable-next-line @typescript-eslint/require-await
 export default async function processExample(
   event: CloudEvent<unknown>,
-  logger: BaseLogger,
+  options: ServiceOptions,
 ): Promise<void> {
   // NB: A production-worthy implementation would do schema validation on `event.data` here.
 
-  logger.info({ event: event.toJSON() }, 'Event processed');
+  options.logger.info({ event: event.toJSON() }, 'Event processed');
```
Comment (gnarea) [inline src/backgroundQueue/sinks/example.sink.ts:11]: Make sure the final event log doesn't include the whole event -- just the few things we'd need to identify the event whilst troubleshooting.

File: src/backgroundQueue/sinks/memberBundleRequestTrigger.sink.spec.ts
Comment (gnarea) [inline src/backgroundQueue/sinks/memberBundleRequestTrigger.sink.spec.ts:1]: We should also test the case where there are no requests to process (no event should be emitted, no DB record deleted).

File: src/backgroundQueue/sinks/memberBundleRequestTrigger.sink.spec.ts [hunk 2]
```diff
@@ -33,14 +61,100 @@ describe('triggerBundleRequest', () => {
...
+    const triggerEvent = new CloudEvent<MemberBundleRequestTriggerPayload>({
+      id: CE_ID,
+      source: CE_SOURCE,
+      type: BUNDLE_REQUEST_TRIGGER_TYPE,
+    });
+
+    await postEvent(triggerEvent, server);
+
+    const publishedEvents = getEvents();
+    expect(publishedEvents).toHaveLength(1);
```
Comment (gnarea) [inline src/backgroundQueue/sinks/memberBundleRequestTrigger.sink.spec.ts:110]: This seems to be outside the scope of this test, and is already tested above.

File: src/backgroundQueue/sinks/memberBundleRequestTrigger.sink.spec.ts [hunk 3]
```diff
@@ -33,14 +61,100 @@ describe('triggerBundleRequest', () => {
...
+  });
+
+  test('Bundle start date more then 3 days into the future should not be sent', async () => {
+    const data = {
+      publicKeyId: MEMBER_PUBLIC_KEY_MONGO_ID,
+      memberBundleStartDate: addSeconds(addDays(new Date(), BUNDLE_REQUEST_DATE_RANGE), 1),
```
Comment (gnarea) [inline src/backgroundQueue/sinks/memberBundleRequestTrigger.sink.spec.ts:143]: A 1-second difference is too small, so this is very likely to be a flaky test. I've been bitten by this too many times on CI, where performance is highly unpredictable 😬 

Let's increase to something like 5 or 10 seconds, which is still a negligible number considering where's dealing with days.

**Review by gnarea — CHANGES_REQUESTED** [review]

File: src/backgroundQueue/sinks/memberBundleRequestTrigger.sink.ts [hunk 4]
```diff
@@ -1,29 +1,56 @@
...
+import type { ServiceOptions } from '../../serviceTypes.js';
 
+export const BUNDLE_REQUEST_DATE_RANGE = 3;
 export default async function triggerBundleRequest(
   event: CloudEvent<unknown>,
-  logger: BaseLogger,
+  options: ServiceOptions,
 ): Promise<void> {
-  logger.info({ source: event.source }, 'Triggering bundle request');
+  options.logger.info({ eventId: event.id, eventType: event.type }, 'Triggering bundle request');
```
[the remark this answers is not in the document]
Comment (levansuper) [reply src/backgroundQueue/sinks/memberBundleRequestTrigger.sink.ts:18]: I think we need to log eventId and eventType to identify the event.

**Pull request merged**